File: solutions/go/prime-factors/3/prime_factors.go
package prime

func Factors(number int64) (factors []int64) {
	for factor := int64(2); number%factor == 0; {
		factors = append(factors, factor)
		number /= factor
	}
	for factor := int64(3); factor*factor <= number; factor += 2 {
		for number%factor == 0 {
			factors = append(factors, factor)
			number /= factor
		}
	}
	if number > 2 {
		factors = append(factors, number)
	}
	return
}

File: solutions/python/prime-factors/3/prime_factors.py
"""
Compute the prime factors of a given natural number.
A prime number is only evenly divisible by itself and 1.
"""


def factors(value: int) -> list[int]:
    """
    Calculate and return the list of prime factors of a given integer.

    Parameters:
    value (int): The integer to factorize.

    Returns:
    list[int]: A list of prime factors of the given integer.
    """
    factor_list = []
    factor = 2

    while value > 1:
        if value % factor:
            factor += 1
        else:
            factor_list.append(factor)
            value /= factor

    return factor_list
